sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -77,34 +78,20 @@ public boolean hasStarted() {
     public void startPushNotifications(String deviceRegistrationToken, Consumer<Throwable> errorHandler) {
         this.deviceRegistrationToken = deviceRegistrationToken;
         this.registrationErrorHandler = errorHandler;
-
+        Log.v("device_id: ", deviceRegistrationToken);
         if (this.isPushNotificationsStarted) {
             return;
         }
-
-        String skypeUserToken;
-        try {
-            skypeUserToken = communicationTokenCredential.getToken().get().getToken();
-        } catch (ExecutionException | InterruptedException e) {
-            throw logger.logExceptionAsError(
-                new RuntimeException("Get skype user token failed for push notification: " + e.getMessage()));
-        }
-
-        try {
-            this.refreshEncryptionKeys();
-            this.registrarClient.register(skypeUserToken, deviceRegistrationToken, this.cryptoKey, this.authKey);
-        } catch (Throwable throwable) {
-            this.logger.error("Failed to start push notifications!");
-            errorHandler.accept(throwable);
-            return;
-        }
+        this.workManager = WorkManager.getInstance();
+        this.registrationDataContainer = RegistrationDataContainer.instance();
```

Comment:
Does registrationDataContainer need to be singleton? If so, why do we need to keep a reference of it?

Reply:
Implementing with a singleton is much safter and easier. The credential keys are loaded from file system and gets refreshed every 15 minutes. The refresh method is invoked by the worker every 15 minutes. So I am using singleton to ensure the same key-store is being shared/used. 

Do you prefer using RegistrationDataContainer.instance() instead of keeping a reference? I just feel having a reference avoid duplicate calls to this method RegistrationDataContainer.instance().